src/internal/connector/graph/middleware.go

```diff
@@ -419,13 +424,55 @@ func extractRateLimiterConfig(ctx context.Context) (LimiterCfg, bool) {
 	return lc, ok
 }
 
+type limiterConsumptionKey string
```

Comment:
nit: Very subjective, but to me this typedef isn't adding much value.

Reply:
It's actually not subjective in this case.  Golang context values are a type of `map[any]any` (where, of course, `any` is an alias for the empty interface, `interface{}`).  Using `any` as a key causes the map to store both the value and the type of the key.

This way it's possible to store equally valued strings without collision, but that only works if both values have a unique type.  Go ahead and try to add to a context using `WithValue` and a plain string: you should see a linter error about keying with an untyped string.  

Bit of an ingenious way to ensure that a shared variable with a generic storage interface stays free of common-name collisions.

Reply:
Thanks! Yeah, this makes total sense.